File: Aigilas/Aigilas/HUD/DeltasHud.cs
using System;
using Aigilas.Creatures;
using Aigilas.Items;
using SPX.Util;
using SPX.Core;

namespace Aigilas.HUD
{
    public class DeltasHud:IHud
    {
        private Equipment _equipment;

        public DeltasHud(ICreature owner, Equipment equipment): base(owner, XnaManager.WindowWidth / 2, XnaManager.WindowHeight / 2)
        {
            _equipment = equipment;
        }

        public void Draw()
        {
            if (_isVisible)
            {
                _textHandler.Draw();
            }
        }

        private GenericItem GetEquipmentIn(int slot)
        {
            if (_equipment.GetItems().ContainsKey(slot))
            {
                return _equipment.GetItems()[slot];
            }
            return null;
        }

        private const string delim = "|";
        private const string positive = "+";
        private const string title = "Deltas";
        private string display = "EMPTY";

        public void Update(GenericItem item,bool refresh)
        {
            if (_isVisible)
            {
                _textHandler.Update();
                _textHandler.Clear();
                if (item != null && refresh)
                {
                    if (GetEquipmentIn(Equipment.ClassToSlot(item.GetItemClass())) != null)
                    {
                        StringSquisher.Clear();
                        foreach (var stat in GetEquipmentIn(Equipment.ClassToSlot(item.GetItemClass())).Modifers.GetDeltas(item.Modifers))
                        {
                            StringSquisher.Squish(((stat > 0) ? positive : String.Empty),StringStorage.Get(stat),delim);
                        }
                        display = StringSquisher.Flush();
                    }
                }
                _textHandler.WriteDefault(title, 30, 260, GetHudOrigin());
                _textHandler.WriteDefault(display, 30, 290, GetHudOrigin());
            }
        }
    }
}

File: aigilas/src/aigilas/hud/DeltasHud.java
package aigilas.hud;

import aigilas.creatures.BaseCreature;
import aigilas.items.Equipment;
import aigilas.items.GenericItem;
import aigilas.items.ItemSlot;
import sps.core.SpxManager;
import sps.util.StringSquisher;
import sps.util.StringStorage;

public class DeltasHud extends BaseHud {
    private final Equipment _equipment;

    public DeltasHud(BaseCreature owner, Equipment equipment) {
        super(owner, SpxManager.WindowWidth / 2, SpxManager.WindowHeight / 2);
        _equipment = equipment;
    }

    public void draw() {
        if (_isVisible) {
            _textHandler.draw();
        }
    }

    private GenericItem getEquipmentIn(ItemSlot slot) {
        if (_equipment.getItems().containsKey(slot)) {
            return _equipment.getItems().get(slot);
        }
        return null;
    }

    private static final String delim = "|";
    private static final String positive = "+";
    private static final String title = "Deltas";
    private String display = "EMPTY";

    public void update(GenericItem item, boolean refresh) {
        if (_isVisible) {
            _textHandler.update();
            _textHandler.clear();
            if (item != null && refresh) {
                if (getEquipmentIn(item.getItemClass().Slot) != null) {
                    StringSquisher.clear();
                    for (Float stat : getEquipmentIn(item.getItemClass().Slot).Modifiers.getDeltas(item.Modifiers)) {
                        StringSquisher.squish(((stat > 0) ? positive : ""), StringStorage.get(stat), delim);
                    }
                    display = StringSquisher.flush();
                }
            }
            _textHandler.writeDefault(title, 30, 260, getHudOrigin());
            _textHandler.writeDefault(display, 30, 290, getHudOrigin());
        }
    }
}
